docs/reference/scripting/access-fields.asciidoc

```diff
@@ -0,0 +1,86 @@
+[[script-fields-api]]
+== Access fields in a document with the `field` API
+++++
+<titleabbrev>Access fields in a document</titleabbrev>
+++++
+
+beta::["The `field` API is still in development and should be considered a beta feature. The API is subject to change and this iteration is likely not the final state.",{es-issue}78920]
+
+Use the `field` API to access document fields:
+
+[source,painless]
+----
+field('my_field').get()
+----
+
+This API fundamentally changes how you access documents in Painless. Previously,
+you had to access the `doc` map with the field name that you wanted to access:
+
+[source,painless]
+----
+doc['my_field'].value
+----
+
+Accessing document fields this way didn't handle missing values or missing
+mappings, which meant that your Painless scripts had to check that fields exist
+and create intermediary objects for handling such omissions.
+
+Instead, use the `field` API, which is the preferred approach to access
+documents in Painless. The `field` API abstracts access options (such as
+`_source` or `doc_values`) and provides a field-centric view of your documents.
```

Comment:
~The `field` API abstracts access options (such as
`_source` or `doc_values`) and provides a field-centric view of your documents.~

The `field` API makes it easy to handle missing values and, in the future, will abstract `_source` and `doc-value` access.

Reply:
I'd like to rewrite this sentence to avoid "in the future", but also provide space for expansion in subsequent iterations:

>The `field` API handles missing values, and will evolve to abstract access to `_source` and `doc_values`.